tests/unit/bash/bash_pkg_conditional_rpm.bats.jinja

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+# Mock the "rpm" command
+# mocked package: coconut-4.5.17-8.fc31
+# The mock is used to avoid the need to have a specific RPM package in a
+# specific version installed on the system where this test is run.
+function rpm()
+{
+    if [[ "$*" == "--quiet -q coconut" ]] ; then
+        return 0
+    elif [[ "$*" ==  "-q --queryformat %{VERSION}-%{RELEASE} coconut" ]] ; then
+        echo "4.5.17-8.fc31"
+        return 0
+    else
+        echo "BATS mock for rpm doesn't support this command"
+        return 1
+    fi
+}
+export -f rpm
+
+@test "bash_pkg_conditional_rpm - test package presence" {
+    {{{ bash_pkg_conditional_rpm("coconut") | indent(4) }}}
+}
+
+@test "bash_pkg_conditional_rpm - test package version" {
+    {{{ bash_pkg_conditional_rpm("coconut", "5", "<") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0", "<") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0.0-1", "<") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0.0-1.fc31", "<") | indent(4) }}}
+
+    {{{ bash_pkg_conditional_rpm("coconut", "5", "<=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0", "<=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0.0-1", "<=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0.0-1.fc31", "<=") | indent(4) }}}
+
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.17-9.fc31", "<") | indent(4) }}}
```

Comment:
I don't like the `.fc31` bit, I think that a content author shouldn't have to worry about it at all.
I am even tempted to suggest to drop the release part - releases stop to be new very fast, then they become common, and even superseded. IOW, if one's system is up-to-date, then the release should be "high enough" at scan-time, as a release that exists at content-writing time will have time to propagate to individual systems.